config/utilities/jsonToNestedValue.test.ts

```diff
@@ -0,0 +1,39 @@
+import {jsonToNestedValue} from './jsonToNestedValue'
+
+describe('Utilities: jsonToNestedValue', () => {
+  const obj = {
+    parent: {
+      subparent: {
+        child1: {
+          value: '#111',
+          $type: 'color'
+        },
+        child2: {
+          value: '#222',
+          $type: 'color'
+        }
+      },
+      otherSubparent: {
+        text: 'hello',
+        number: 42,
+        noValue: {
+          string: 'string',
+          number: 12
+        }
+      }
+    }
+  }
+
+  it('it returns only value property as value of object', () => {
+    expect(jsonToNestedValue(obj['parent']['subparent']['child1'])).toStrictEqual('#111')
```

Comment:
just curious.. do you need to use index here? makes it a little harder to read... this should work same...

```suggestion
   expect(jsonToNestedValue(obj.parent.subparent.child1)).toBe('#111')
```